torch/_export/__init__.py

```diff
@@ -334,7 +334,7 @@ def aot_compile(
             restore_fqn=False,
         )
 
-    with torch.no_grad():
+    with torch.no_grad(), compile_context(gm.meta.get('dynamo_compile_context', None)):
```

Comment:
This seems like a suboptimal place to put it, because if there is a call to `aot_compile` that doesn't go through this function you won't have compile context set